pyupgrade.py

```diff
@@ -1240,12 +1320,16 @@ def visit_If(self, node):  # type: (ast.If) -> None
 
     def visit_For(self, node):  # type: (ast.For) -> None
         if (
-            not self._in_async_def and
-            len(node.body) == 1 and
-            isinstance(node.body[0], ast.Expr) and
-            isinstance(node.body[0].value, ast.Yield) and
-            targets_same(node.target, node.body[0].value.value) and
-            not node.orelse
+                not self._in_async_def and
+                len(node.body) == 1 and
+                isinstance(node.body[0], ast.Expr) and
+                isinstance(node.body[0].value, ast.Yield) and
+                targets_same(
+                    node.target,
+                    # node.body[0].value.value cannot be None here
```

Comment:
it actually can be, interesting...

```console
$ pyupgrade t.py  --py3-plus
Traceback (most recent call last):
  File "/private/tmp/venv/bin/pyupgrade", line 10, in <module>
    sys.exit(main())
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1830, in main
    ret |= fix_file(filename, args)
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1804, in fix_file
    contents_text = _fix_py3_plus(contents_text)
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1503, in _fix_py3_plus
    visitor.visit(ast_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1244, in generic_visit
    super(FindPy3Plus, self).generic_visit(node)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 270, in generic_visit
    self.visit(item)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1089, in _visit_sync_func
    self._visit_func(node)
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1085, in _visit_func
    self.generic_visit(node)
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1244, in generic_visit
    super(FindPy3Plus, self).generic_visit(node)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 270, in generic_visit
    self.visit(item)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 1235, in visit_For
    targets_same(node.target, node.body[0].value.value) and
  File "/private/tmp/venv/lib/python3.7/site-packages/pyupgrade.py", line 957, in targets_same
    for t1, t2 in zip(ast.walk(target), ast.walk(yield_value)):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 234, in walk
    todo.extend(iter_child_nodes(node))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 189, in iter_child_nodes
    for name, field in iter_fields(node):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 177, in iter_fields
    for field in node._fields:
AttributeError: 'NoneType' object has no attribute '_fields'
$ cat t.py 
def f():
    for x in y:
        yield
```

Reply:
Hmmmm the cast here was based on the current test suite. I can add a test and do the right thing :tm: